lib/matplotlib/sphinxext/plot_directive.py

```diff
@@ -775,14 +830,22 @@ def run(arguments, content, options, state_machine, state, lineno):
             function_name = None
 
         code = Path(source_file_name).read_text(encoding='utf-8')
-        output_base = os.path.basename(source_file_name)
+        if options['output-base-name']:
+            output_base = options['output-base-name']
+            check_output_base_name(env, output_base)
+        else:
+            output_base = os.path.basename(source_file_name)
     else:
         source_file_name = rst_file
         code = textwrap.dedent("\n".join(map(str, content)))
-        counter = document.attributes.get('_plot_counter', 0) + 1
-        document.attributes['_plot_counter'] = counter
         base, ext = os.path.splitext(os.path.basename(source_file_name))
-        output_base = '%s-%d.py' % (base, counter)
+        if options['output-base-name']:
+            output_base = options['output-base-name']
```

Comment:
Don't we need a .py extension here like in the else case?

Reply:
This block handles the case where there isn't an extension. https://github.com/matplotlib/matplotlib/pull/28187/files#diff-1565d78306cf72de5a28f4e2b196ee2cb09026d1f8aa2dc26178c3a47f3347b8R853-R856